.github/workflows/benchmarks.yaml

```diff
@@ -0,0 +1,78 @@
+name: Benchmarks
+on:
+  pull_request_target:
+    branches:
+      - master
+
+concurrency:
+  group: ${{ format('{0}-{1}', github.workflow_ref, github.head_ref) }}
+  cancel-in-progress: true
+
+jobs:
+  benchmarks:
+    if: github.repository_owner == 'mybatis'
+    permissions:
+      contents: read
+      pull-requests: write # for benchmark comment
+    runs-on: ubuntu-latest
+    env:
+      COMMENT_FILE: ${{ github.workspace }}/benchmark-comment.md
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - id: get-sha
+        run: |
+          # getting exact master sha at the moment to reference it in the comment
+          echo "sha=$( curl -u "u:${{ github.token }}" https://api.github.com/repos/${{ github.repository }}/git/ref/heads/${{ github.base_ref }} | jq .object.sha | tr -d '"' )" >> "$GITHUB_OUTPUT"
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ steps.get-sha.outputs.sha }}
+          path: benchmark-master
+
+      - uses: actions/checkout@v4
+        with:
+          path: benchmark-pull-request
```

Comment:
FYI we also started using `pull_request_target` recently and one difference we found is that for this event, the `ref` has to be explicit:
https://github.com/Breus/json-masker/blob/master/.github/workflows/benchmark.yml#L34

Reply:
Thanks for this! ill update :)

Reply:
Shamelessly added all of your updates in here ;-)